libr/core/cmd.c

```diff
@@ -996,8 +997,55 @@ static int cmd_kuery(void *data, const char *input) {
 	const char *sp, *p = "[sdb]> ";
 	const int buflen = sizeof (buf) - 1;
 	Sdb *s = core->sdb;
+	
+	char *cur_pos, *cur_cmd, *next_cmd;
+	char *temp_pos, *temp_cmd, *temp_storage;
 
 	switch (input[0]) {
+
+	case 'j':
+		out = sdb_querys (s, NULL, 0, "anal/**");
+
+		r_cons_printf ("{\"anal\":{");
+
+		while (strlen (out) != 0) {
+			cur_pos = strchr (out, '\n');
+			cur_cmd = strndup (out, cur_pos - out);
+
+			r_cons_printf ("\n\n\"%s\" : [", cur_cmd);
+
+			next_cmd = malloc (strlen (cur_cmd) + 8);
+			// malloc for this: "anal/" + (input + 1) + cur_cmd + "/*"
+
+			strcpy (next_cmd, "anal/");
+			strcat (next_cmd, cur_cmd);
+			strcat (next_cmd, "/*");
+
+			temp_storage = sdb_querys (s, NULL, 0, next_cmd);
+
+			if (!temp_storage) {
+				r_cons_println ("\nEMPTY\n");
+				r_cons_printf ("],\n\n");
+				out += cur_pos - out + 1;
+				continue;
+			}
+
+			while (strlen (temp_storage) != 0) {
+				temp_pos = strchr (temp_storage, '\n');
```

Comment:
strchr can return  null and make r2 crash

Reply:
Will handle the null case for strchr and make other changes too.